src/admin/index.js

```diff
@@ -418,6 +418,39 @@ module.exports = ({ retry = { retries: 5 }, logger: rootLogger, cluster }) => {
     })
   }
 
+  /**
+   * Retrieve metadata for a topic.
+   *
+   * @param {string} topic
+   * @return {Promise<TopicMetadata>}
+   *
+   * @typedef {Object} TopicMetadata
+   * @property {String} name
+   * @property {Array<PartitionMetadata>} partitions
+   *
+   * @typedef {Object} PartitionMetadata
+   * @property {number} partitionErrorCode Response error code
+   * @property {number} partitionId Topic partition id
+   * @property {number} leader  The id of the broker acting as leader for this partition.
+   * @property {Array<number>} replicas The set of all nodes that host this partition.
+   * @property {Array<number>} isr The set of nodes that are in sync with the leader for this partition.
+   *
+   * @see https://kafka.apache.org/protocol#The_Messages_Metadata
+   */
+  const getTopicMetadata = async topic => {
```

Comment:
Can you change this to accept any number of topics?

```javascript
admin.getTopicMetadata({
  topics: ['topic-A', 'topic-B']
})
```

The use of an object seems strange to begin with, but it allows us to extend the API later, for example, adding a timeout option, etc.

The API should also allow you to fetch metadata for all topics, refresh metadata will do that if no topics are included with the target topic API.

```javascript
admin.getTopicMetadata()
```

Reply:
Seems like a sensible design choice to me @tulios. I had the method return an object with property `partitions` rather than the partitions array itself for similar reasons. I'll push that change today.